### PR TITLE
Added Cypress test for validating MM-18039 Members Invitation Flow: Verify UI Elements

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
Validating the UI Elements of the Members Invitation Flow when Guest Account feature is enabled. Please check the ticket for more details. 

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18039

### CODE REVIEW

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        cy.getByTestId('invitationModal').within(($el) => {
+            cy.wrap($el).find('h1').should('have.text', `Invite people to ${teamName}`);
+        });
+        cy.getByTestId('inviteMembersLink').should('be.visible').within(($el) => {
+            cy.wrap($el).getByTestId('inviteMembersSection').should('have.text', 'Invite MembersInvite new team members with a link or by email. Team members have access to messages and files in open teams and public channels.');
+            cy.wrap($el).find('.arrow').click();
+        });
+
+        // *Verify Share Link field
+        cy.getByTestId('shareLinkInput').should('be.visible').and('not.have.value', '');
```
Comment (saturninoabril) [inline e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: I think we should actually test the value of the text, like:
```suggestion
        cy.getByTestId('shareLinkInput').should('be.visible').and('have.value', `${baseUrl}/signup_user_complete/?id=${inviteId}`);
```
Where:
- `baseUrl` from Cypress.config('baseUrl');`
- `inviteId` from getting team by Id.  May add new API custom command of `cy.apiGetTeam`
```javascript
/**
 * Gets a team on the system
 * * @param {String} teamId - The team ID to get
 * All parameter required
 */

Cypress.Commands.add('apiGetTeam', (teamId) => {
    return cy.request({
        headers: {'X-Requested-With': 'XMLHttpRequest'},
        url: `api/v4/teams/${teamId}`,
        method: 'GET',
    }).then((response) => {
        expect(response.status).to.equal(200);
        return cy.wrap(response);
    });
});
```

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        // *Verify Invite People field
+        cy.get('#inviteMembersButton').scrollIntoView().should('be.visible').and('be.disabled');
+        cy.getByTestId('inputPlaceholder').should('be.visible').within(($el) => {
+            // # Type the email of the new user
+            cy.wrap($el).get('input').type(email, {force: true});
+            cy.wrap($el).get('.users-emails-input__menu').
+                children().should('have.length', 1).
+                eq(0).should('contain', `Invite ${email} as a team member`).click();
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Can we have more UI check on Invite Members Modal?
Screen Shot 2019-09-13 at 6 30 36 PM

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        // * Verify the behavior when Cancel button in the confirmation message is clicked
+        cy.get('#cancelModalButton').click();
+        cy.get('#confirmModal').should('not.exist');
+
+        // * Verify the behavior when Yes, Discard button in the confirmation message is clicked
+        cy.get('#closeIcon').should('be.visible').click();
+        cy.get('#confirmModalButton').should('be.visible').and('have.text', 'Yes, Discard').click();
+        cy.get('.InvitationModal').should('not.exist');
+    });
+});
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Sure. I will make the necessary changes

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Sure. I will make the necessary changes

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    before(() => {
+        // # Enable Guest Account Settings
+        cy.apiUpdateConfig({
+            GuestAccountsSettings: {
+                Enable: true,
+            },
+        });
+
+        // # Login as "sysadmin" and go to /
+        cy.apiLogin('sysadmin');
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Sure. I will add that as a separate testcase and PR

File: components/invitation_modal/invitation_modal_initial_step.jsx [hunk 6]
```diff
@@ -46,13 +47,15 @@ export default class InvitationModalInitialStep extends React.Component {
                         <FormattedMessage
                             id='invitation_modal.invite_members.description'
                             defaultMessage='Invite new team members with a link or by email. Team members have access to messages and files in open teams and public channels.'
+                            data-testid='inviteMembersHelper'
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply components/invitation_modal/invitation_modal_initial_step.jsx]: Actually I had made this change, but the data-testid is not rendered in the UI in this span. I have changed the assertion. Will remove this change

File: components/invitation_modal/invitation_modal_initial_step.jsx [hunk 7]
```diff
@@ -64,6 +67,7 @@ export default class InvitationModalInitialStep extends React.Component {
                         <FormattedMessage
                             id='invitation_modal.invite_guests.description'
                             defaultMessage='Invite guests to one or more channels. Guests only have access to messages, files, and people in the channels they are members of.'
+                            data-testid='inviteGuestHelper'
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply components/invitation_modal/invitation_modal_initial_step.jsx]: Actually I had made this change, but the data-testid is not rendered in the UI in this span. I have changed the assertion. Will remove this change

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Changes done.

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Changes done.

File: components/invitation_modal/invitation_modal_initial_step.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (srkgupta) [reply components/invitation_modal/invitation_modal_initial_step.jsx]: Changes done.

File: components/invitation_modal/invitation_modal_initial_step.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (srkgupta) [reply components/invitation_modal/invitation_modal_initial_step.jsx]: Changes done.

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        // #Open Invite People
+        cy.get('#sidebarHeaderDropdownButton').should('be.visible').click();
+        cy.get('#invitePeople').should('be.visible').click();
+
+        // *Verify UI Elements in initial step
+        cy.getByTestId('invitationModal').within(($el) => {
+            cy.wrap($el).find('h1').should('have.text', `Invite people to ${teamName}`);
+        });
+        cy.getByTestId('inviteMembersLink').should('be.visible').within(($el) => {
+            cy.wrap($el).getByTestId('inviteMembersSection').should('have.text', 'Invite MembersInvite new team members with a link or by email. Team members have access to messages and files in open teams and public channels.');
```
[the remark this answers is not in the document]
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Changes done.

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js (hunk 1 above)
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Changes done.

File: e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js (hunk 2 above)
Comment (srkgupta) [reply e2e/cypress/integration/guest_accounts/member_invitation_ui_spec.js]: Changes done.

**Comment by thekiiingbob** [comment]
Approved, but more linting errors to fix @srkgupta .

**Comment by srkgupta** [comment]
Actually my windows is having troubles with eslint checks. Let me check.

**Pull request merged**